File: cpp/algoDS/reverseString/TestReverseString.cpp
#include <stdio.h>
#include <string.h>

void reverseString(char* word);

int main() {
    char word[4] = "abc";
    reverseString(word);
    return 0;
}

void reverseString(char* word) {
    printf("reverseString: input word=[%s]\n", word);
    if (word == NULL) {
	printf("[E] word == NULL !!!");
	return;
    }
    int len = strlen(word);
    if (len <= 1) {
        printf("[W] len <= 1. No need to reverse !");
	return;
    }
    for (int i = 0; i < len/2; i++) {
        char temp = word[i];
	word[i] = word[len-1-i];
	word[len-1-i] = temp;
	i++;
    }
    printf("reverseString: output word=[%s]\n", word);
}

File: java/algoDS/reverseString/TestReverseString.java
public class TestReverseString {
    public static void main(String[] args) {
        String inWord = "abc";
        String outWord = reverseString(inWord);
    }

    public static String reverseString(String inWord) {
        String outWord = null;

        if (inWord != null) {
            System.out.println("[D] reverseString: inword=[" + inWord + "]");
        } else {
            System.out.println("[E] reverseString: inWord == null !!!");
            return outWord;
        }

        int len = inWord.length(); 
        if (len <= 1) {
            System.out.println("[I] reverseString: len <= 1. No need to reverse.");
            outWord = inWord;
            return outWord;
        }

	char[] inWordArray = inWord.toCharArray();
        for (int i = 0; i < len/2; i++) {
            char temp = inWord.charAt(i);
	    inWordArray[i] = inWordArray[len-1-i];
	    inWordArray[len-1-i] = temp;
	    i++;
        }

	outWord = new String(inWordArray);
        System.out.println("[V] reverseString: outWord=[" + outWord + "]");

        return outWord;
    }
}

